packages/react-server/core/components/RootElement.js

```diff
@@ -103,7 +103,7 @@ RootElement.getRootElementAttributes = function(element) {
 		'id',
 		'style',
 	].forEach(k => props[k] && (attrs[k] = props[k]));
-
+	attrs.tagName = props.tagName || 'div';
```

Comment:
Yea, I was waffling a bit on that one, flipped a coin and came out this way.  I think I agree with you that the other way is easer to read.